### ISSUE TITLE
Generated Immutable classes don't seem to have the @Immutable or @Generated annotations

### ISSUE THREAD

**Issue by kanesee** [immutables/immutables#1453]
When I look at the generated immutables class for a Pojo such as

```
@Value.Immutable
@JsonSerialize(as = ImmutableMyPojo.class)
@JsonDeserialize(as = ImmutableMyPojo.class)
public interface MyPojo {
}
```

I see something like this generated

```
@Generated(from = "MyPojo", generator = "Immutables")
@SuppressWarnings({"all"})
@SuppressFBWarnings
@ParametersAreNonnullByDefault
@javax.annotation.processing.Generated("org.immutables.processor.ProxyProcessor")
@Immutable
@CheckReturnValue
public final class ImmutableMyPojo
    implements MyPojo {
}
```

It seems to have the `@Generated` and `@Immutable` annotations.

But when I check them with `isAnnotationPresent`, it's not there

```
ImmutableMyPojo.class.isAnnotationPresent(org.immutables.value.Generated.class)
```

When I call `ImmutableMyPojo.class.getAnnotations()`, only `@ParametersAreNonnullByDefault` and `@CheckReturnValue` are in the annotations.

I want to be able to detect if a class is an immutable.
What am I missing?

**Comment by kanesee** [comment]
Digging a bit further, it seems to be because the [@Generated](https://docs.oracle.com/javase/7/docs/api/javax/annotation/Generated.html) annotations `@Retention(value=SOURCE)` which means it shows up at compile-time but not runtime :(

Is there any other way to determine if the class is an Immutables class?

**Comment by kanesee** [comment]
Digging further, seems it might be because the default [@RetentialPolicy](https://docs.oracle.com/javase/7/docs/api/java/lang/annotation/RetentionPolicy.html#CLASS) on `@Generated` and `@Immutable` is `CLASS`

> Annotations are to be recorded in the class file by the compiler but need not be retained by the VM at run time. This is the default behavior.

I suppose unless I write a custom AnnotationProcessor, I won't be able to know whether a class is an Immutable or not.

Not sure if others have a need to know this info at runtime, but maybe a future version update to change the RetentionPolicy of one of the annotations above

**Comment by elucash** [comment]
Yes, it's retention. There's annotation-injection functionality available, in org.immutables:annotate . Once I get back to my laptop I can send an example how to use it. In a meanwhile you can try to read javadocs on InjectAnnotation

**Comment by elucash** [comment]
In addition, there's custom immutable annotation, see guide/docs http://immutables.github.io/style.html#custom-immutable-annotation Obviously, this might be helpful only to declutter and streamline your definitions, by allowing to combine styles and aforementioned annotation injections.
